logserver/src/test/java/alluxio/logserver/AlluxioLog4jSocketNodeTest.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.logserver;
+
+import static alluxio.logserver.AlluxioLog4jSocketNode.setAcceptList;
+
+import org.apache.commons.io.serialization.ValidatingObjectInputStream;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.log4j.config.PropertySetterException;
+import org.apache.log4j.helpers.UtilLoggingLevel;
+import org.apache.log4j.lf5.Log4JLogRecord;
+import org.apache.log4j.lf5.LogLevel;
+import org.apache.log4j.lf5.LogLevelFormatException;
+import org.apache.log4j.lf5.util.AdapterLogRecord;
+import org.apache.log4j.lf5.viewer.LogTableColumn;
+import org.apache.log4j.lf5.viewer.LogTableColumnFormatException;
+import org.apache.log4j.pattern.LogEvent;
+import org.apache.log4j.spi.LocationInfo;
+import org.apache.log4j.spi.LoggingEvent;
+import org.apache.log4j.spi.ThrowableInformation;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.util.Hashtable;
+
+public class AlluxioLog4jSocketNodeTest {
+
+  private ByteArrayOutputStream mByteArrayOutputStream;
+  private ObjectOutputStream mObjectOutputStream;
+  byte[] mBuffer;
+  private ByteArrayInputStream mByteArrayInputStream;
+  private ValidatingObjectInputStream mValidatingObjectInputStream;
+
+  private void createOutputStreams() throws IOException {
+    mByteArrayOutputStream = new ByteArrayOutputStream(100);
+    mObjectOutputStream = new ObjectOutputStream(mByteArrayOutputStream);
+  }
+
+  private void write2BufferAndCloseOutputStream() throws IOException {
+    mBuffer = mByteArrayOutputStream.toByteArray();
+    mObjectOutputStream.close();
+    mByteArrayOutputStream.close();
+  }
+
+  private void createInputStreamsAndsetAcceptList() throws IOException {
+    mByteArrayInputStream = new ByteArrayInputStream(mBuffer);
+    mValidatingObjectInputStream = new ValidatingObjectInputStream(mByteArrayInputStream);
+    setAcceptList(mValidatingObjectInputStream);
+  }
```

Comment:
How about a single method to cover all these things?

```java
<T> serializeThenDeserialize(T object) throws SomeException {
  create object output stream
  write object into stream
  get byte buffer out of output stream
  create object input stream from byte buffer
  set accept list
  read object from stream
  return object
}
```

Also properly handle exceptions when dealing with the streams.